transcom/mymove
# MB-3157 Handle distance out of bounds for pricing DLH and DSH

## Description

We discovered that `Zip5TransitDistance` and `Zip3TransitDistance` had checks in them for distance minimums but those checks do not apply to all places we need to call `Zip5TransitDistance` and `Zip3TransitDistance` from. This PR removes the check from `Zip3TransitDistance` since it hasn't been used yet, and introduces a rename to `Zip5TransitDistance` since it has a check for line haul distance minimum and adds a `Zip5TransitDistance` that doesn't do the check.

This PR didn't move the check out of the original `Zip5TransitDistance` as it's called in a lot of places and it wasn't always clear if the error should be checked and returned there or not. I would discourage further use of the new `Zip5TransitDistanceLineHaul` in favor of not putting the checks in the planner code. 

## Reviewer Notes

Is the updated wiki page mentioned below clear?

## Setup

To test locally

```sh
make db_dev_e2e_populate
make server_run
make client_run
```

Follow the steps in the [acceptance testing payment requests wiki page](https://github.com/transcom/mymove/wiki/Acceptance-Testing-Payment-Requests#creating-a-new-mto-with-shipments) to setup a brand new MTO with a DLH and DSH.

* Once the MTO is sent to the prime, and the shipments are approved
* Update the actual weight on both shipments (no need to update estimated weight it will already be set)
* Create a payment request for both DLH and DSH services.
* Verify the price is correct

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/backend#querying-the-database-safely)
 have been satisfied.

## References

* [MB-3157](https://dp3.atlassian.net/browse/MB-3157) for this change
* [this wiki page](https://github.com/transcom/mymove/wiki/Acceptance-Testing-Payment-Requests) explains more about the approach used.
